src/Raven.Server/ServerWide/Commands/Subscriptions/RecordBatchSubscriptionDocumentsCommand.cs

```diff
@@ -104,16 +107,12 @@ public override unsafe void Execute(ClusterOperationContext context, Table items
                     context.ReadObject(existingValue, subscriptionName);
                     shouldUpdateChangeVector = false;
                 }
-                
-                if (subscriptionState.ChangeVectorForNextBatchStartingPoint != PreviouslyRecordedChangeVector)
-                {
-                    throw new SubscriptionChangeVectorUpdateConcurrencyException($"Can't record subscription with name '{subscriptionName}' due to inconsistency in change vector progress. Probably there was an admin intervention that changed the change vector value. Stored value: {subscriptionState.ChangeVectorForNextBatchStartingPoint}, received value: {PreviouslyRecordedChangeVector}");
-                }
+
+                CheckConcurrencyForBatchCv(subscriptionState.ChangeVectorForNextBatchStartingPoint, subscriptionName);
 
                 if (shouldUpdateChangeVector)
                 {
-                    subscriptionState.ChangeVectorForNextBatchStartingPoint =
-                        ChangeVectorUtils.MergeVectors(CurrentChangeVector, subscriptionState.ChangeVectorForNextBatchStartingPoint);
+                    subscriptionState.ChangeVectorForNextBatchStartingPoint = ChangeVectorUtils.MergeVectors(CurrentChangeVector, subscriptionState.ChangeVectorForNextBatchStartingPoint);
```

Comment:
we need to have a separte state per each shard, which also implies that all of those commands that changed the base class should actually be new ones? create an abstraction?